### DISCUSSION TITLE
listsAsArrays true but value doesn't update as Array

### DISCUSSION THREAD

**Discussion by koseenes** [react-querybuilder/react-querybuilder#592] [Q&A] [answered]
Hi,
I use dynamic value editor, when i choose "=" or "!=" operator select control is single select, when i choose "in" or "not in" select control is multiselect. 

Now i need when i choose "=" or "!=" operator, value have to be primitive, when i choouse "in" or "not in" operator, value have to be array. Actually is working but when i change select component to multiselect component, value doesn't change by array, it changes after my selection. Is it bug?

You can see below example, if you change operator from "=" to "in", you will see value still primitive, but it have to be array because listsAsArray is true.
https://codesandbox.io/p/sandbox/cocky-goldberg-nqrshg?file=%2Fsrc%2FApp.tsx%3A56%2C41

**Comment by jakeboone02** [comment, marked as answer]
`listsAsArrays` only has an effect when the `value` is updated. It won't update the `value` during render.

If you want to manipulate rule values before rendering, try [`transformQuery`](https://react-querybuilder.js.org/docs/utils/misc#transformquery).
